product_docs/docs/pgd/5/manual/04-installing-software.mdx

```diff
@@ -0,0 +1,352 @@
+This command format (`echo ... | sudo ... tee -a ...`) appends the echoed string to the end of the postgresql.conf file, which is owned by another user.
+!!!
+
+#### Set the `wal_level`
+
+The BDR extension needs to set the server to perform logical replication. We do this by setting `wal_level` to `logical`.
+
+```
+echo -e "wal_level = 'logical'" | sudo -u enterprisedb tee -a /var/lib/edb/as16/data/postgresql.conf >/dev/null
+
+```
+
+#### Enable commit timestamp tracking
+
+The BDR extension also needs the commit timestamp tracking enabled.
+
+```
+echo -e "track_commit_timestamp = 'on'" | sudo -u enterprisedb tee -a /var/lib/edb/as16/data/postgresql.conf >/dev/null
+
+```
+
+#### Raise `max_worker_processes`
+
+To communicate between multiple nodes, Postgres Distributed nodes run more worker processes than usual.
+The default limit (8) is too low even for a small cluster.
+
+Raise the maximum number of worker processes with this commmand:
+
+```
+echo -e "max_worker_processes = '16'" | sudo -u enterprisedb tee -a /var/lib/edb/as16/data/postgresql.conf >/dev/null
+
+```
+
+The `max_worker_processes` value should be raised for larger clusters.
+To calculate the needed value see [Postgres configuration/settings](../postgres-configuration/#postgres-settings).
+
+#### Add a password to the Postgres enterprisedb user
+
+To allow connections between nodes, a password needs to be set on the Postgres enterprisedb user. 
+For this example, we are using the password `secret`.
+Select a different password for your deployments.
+You will need this password when we get to [Connecting the Cluster](05-connecting-cluster).
+
+```
+sudo -u enterprisedb psql edb -c "ALTER USER enterprisedb WITH PASSWORD 'secret'"
+
+```
+
+#### Enable inter-node authentication in pg_hba.conf
+
+Out of the box, Postgres allows local authentication and connections with the database but not external network connections. 
+To enable this, edit `pg_hba.conf` and add appropriate rules, including rules for the replication users.
+To simplify the process, use this command:
+
+```
+echo -e "host all all all md5\nhost replication all all md5" | sudo tee -a /var/lib/edb/as16/data/pg_hba.conf
+
+```
+
+It will append
+
+```
+host all all all md5
+host replication all all md5
+
+```
+
+to `pg_hba.conf` which will enable the nodes to replicate.
+
+#### Enable authentication between nodes
+
+As part of the process of connecting nodes for replication, PGD logs into other nodes. 
+It will perform that log in as the user that Postgres is running under.
+For epas, this is the `enterprisedb` user.
+That user will need credentials to log into the other nodes.
+We will supply these credentials using the `.pgpass` file which needs to reside in the user's home directory.
+The home directory for `enterprisedb` is `/var/lib/edb`.
+
+Run this command to create the file:
+
+```
+echo -e "*:*:*:enterprisedb:secret" | sudo -u enterprisedb tee /var/lib/edb/.pgpass; sudo chmod 0600 /var/lib/edb/.pgpass
```

Comment:
We could just say that people should ensure that the password for the replication user is documented inside `.pgpass`, and link the relevant section of the PostgreSQL docs.

Reply:
We could, but we aren't. I will add a link to the .pgpass docs though.